cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceBatchContext.java

```diff
@@ -163,4 +169,25 @@ private Output getOutput(Output output) {
     }
     return output;
   }
+
+  @Override
+  public MessagePublisher getMessagePublisher() {
+    // TODO: use caller?
```

Comment:
should use it, it's used to unset MDC keys in the logger so system logs are not tagged with pipeline MDC keys